Add database locked error handling and new function to support stashapp-tag-importer changes.

unatco90 opened the pull request:
This PR makes two changes that are required to support changes in [soundchaser128/
stashapp-tag-importer](https://github.com/soundchaser128/stashapp-tag-importer/pull/1):
- Adds error handling for when the database is locked.
  - When GraphQL's database is locked, it returns a response code of 200 which is "normal," but the GraphQL query actually fails.
  - This change looks for the phrase "database is locked" in the response message and then manually triggers an error.
  - Since it produces an error, applications use their own error handling to retry API calls that fail due to database locks.
- Adds a find_scene_markers_filter() function for detailed marker search.
  - The find_scene_markers() only takes a scene ID, this functions takes a SceneMarkerFilterType dict which can filter markers by tag, performer, etc. 
  - This is used heavily by stashapp-tag-importer to find markers and update / propagate associated tags.
  - Technically, find_scene_markers() could be updated to take the above SceneMarkerFilterType dict, but I didn't want to break any existing things that use the ID only input for that function.

unatco90 committed:
Add error handling for when database is locked.

When GraphQL's database is locked, it returns a response code of 200
which is "normal", however when the database is locked, the GraphQL
query fails. This change is necessary to allow handling of these cases.

unatco90 committed:
Add find_scene_markers_filter() function for detailed marker search.

The find_scene_markers() only takes a scene ID, this functions takes a
SceneMarkerFilterType dict which can filter markers by tag, performer,
etc. This is used heavily by stashapp-tag-importer to find markers and
update / propagate associated tags.

stg-annon commented:
I think we should break `find_scene_markers()` or deprecate it a simple check on the first arg to shunt the request off to the deprecated function with a warning, Ideally the functions line up with the api ones so there is as little confusion as possible

I would suggest renaming  `find_scene_markers()` to `get_scene_markers()` and the new function would take its place checking for a dict as the filter argument then pass it along to the old function with a warning about the change

I would have to assume there was not yet a findSceneMarkers function when `find_scene_markers()` was introduced otherwise I would have used that instead of findScene


Edit:
Looking further at `SceneMarkerFilterType` it seems to not completely replace just getting the markers for a specific scene so `get_scene_markers()` would likely stick around

the dict check would probably look like
```py

if not isinstance(scene_marker_filter, dict):
	self.log.warning("find_scene_markers() no longer accepts scene_id, use get_scene_markers() instead")
	return self.get_scene_markers(scene_marker_filter)

```

unatco90 committed:
Update find_scene_markers_filter() to find_scene_markers().

- Rename find_scene_markers() to get_scene_markers().
- Rename find_scene_markers_filter() to find_scene_markers().
- Add check in find_scene_markers() to redirect to get_scene_markers()
  when legacy find_scene_markers() calls are detected.

unatco90 commented:
Updated with your feedback. I agree with the breaking change as well, and your dict check is a great way of handling legacy calls to find_scene_markers().

stg-annon committed:
update internal refences to find_scene_markers() + full docstring of get_scene_markers()
